### DISCUSSION TITLE
Reorganization of `expect()`, `amGoingTo()`, `comment()`, `wantTo()`

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [Codeception/Codeception#6146] [Ideas]
Currently, there are 4 "comment" methods that all do (almost) the same:

| Code | Output (`-vvv` or `--steps`) |
| -- | -- |
| $I->expect('foo'); | I expect foo |
| $I->expectTo('foo'); | I expect to foo |
| $I->amGoingTo('foo'); | I am going to foo |
| $I->comment('foo'); | foo |

So I'm suggesting to drop the first 3 in favor of **`comment()`**. Reasons:
* The term `comment` is self-explanatory in the sense that it gives you a hint where/how this will be displayed. `amGoingTo` and `expectTo` are non-saying.
* Having 3 methods just for prefixing the comment with 2, 3, or 4 single words (e.g. "I expect") is ridiculous. If you want the exact same output, you can do `$I->comment('I expect foo');`
* "expect" is easily confused with "assert", see https://github.com/Codeception/Codeception/issues/6076#issuecomment-762260883

And then there's `wantTo()` which is inconsistently displayed when called *multiple* times:
```php
$I->wantTo('foo');
$I->wantTo('bar');
```
... leads to:
> ✔ AppCest: Bar (0.04s)

> App\Tests.functional Tests (1) -----------------------------------------------------------------------------------------------------------------------------------------------------------------
AppCest: Foo

Having some way to add a comment to the test overview output is very handy. But `wantTo` is semantically the wrong term for it; and the `$I->` syntax leads to unanswerable questions when called multiple times.
Suggestions? Maybe an annotation above the function instead of a method within the function?

This would resolve https://github.com/Codeception/Codeception/issues/6076 and https://github.com/Codeception/Codeception/issues/6075

**Comment by lon9man** [comment]
how to use these methods in Unit testing? 
PHPStorm doesn't show any comments produced be these methods

**Reply by Naktibalda** [reply]
Don't use these methods in unit tests.

**Reply by lon9man** [reply]
@Naktibalda 
ok. 

for example: 
in PHPStorm Codeception output has slightly different look than PHPUnit. as minimum PHPUnit groups tests if they have input data using @dataProvider, but doesn't show each step as Codeception does. 
![image](https://github.com/Codeception/Codeception/assets/10497794/c350e49c-7e28-4e5a-b63c-43028917bb41)
![image](https://github.com/Codeception/Codeception/assets/10497794/d989b5e8-9f5d-4fbe-b8d1-20e0d9bd7b6a)


question: 
1. does exist any way to add comments to Unit tests in Codeception, which will be visible on runtime?
2. is it possible to change or prevent text of some steps, which AFAIU builded automatically dependently to which actor's method was called?

thanks!

**Reply by lon9man** [reply]
1.5 years passed, but question is still the same:
how to add some comment to codeception steps? saying i want to show text, json, variable and etc.

PHPStorm has 2 panels for testing results:
1. steps (left-side)
2. errors (right-side)
<img width="1881" alt="image" src="https://github.com/user-attachments/assets/02ce7452-8358-43e0-8459-b3b9fb56268c" />

when step is successful - then right panel is empty, but in this case it will be very useful to place there some information about step. developer knows what he wants to show (text, json, variable and etc).

thanks!
